### ISSUE TITLE
[Bug]: fail to get shard leaders from QueryCoord: no replica available

### ISSUE THREAD

**Issue by SKV1983** [milvus-io/milvus#18205]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster): Cluster
- SDK version(e.g. pymilvus v2.0.0rc2): 2.0.2
- OS(Ubuntu or CentOS): Ubuntu
- CPU/Memory: 64
- GPU: 
- Others: 

We set a cluster of 7 query nodes. A collection with 9 Shards of 21M entities with 1024 DIM were inserted. IVF_SQ8 Index used.
```

### Current Behavior

**ISSUE 1**  

The cluster was working fine, like 

roughly 400 search requests per min.

**Search latency**
300 ms 75 percentile.
400 ms 95 percentile.
500 ms 99 percentile.

But after some time, we see the search latency has increased for each request, hence the lower throughput, 60 search requests per minute.

**Search latency**
2 secs 75 percentile
3.75 secs 95 percentile
4 secs 99 percentile.

We could see one query node is taking all the requests, i.e one query node is under 60% CPU Utilization and others around 6%, but RAM usage across all of them was same (7-8 GB).

Since, its a PROD system, we restarted the entire cluster in a hope to make it available, from then we are facing **ISSUE 2**, upon calling `search()` now its throwing following error,

```
RPC error: [_execute_search_requests], <BaseException: (code=1, message=fail to search on all shard leaders, err=All attempts results:
attempt #1:fail to get shard leaders from QueryCoord: no replica available
attempt #2:fail to get shard leaders from QueryCoord: no replica available
attempt #3:fail to get shard leaders from QueryCoord: no replica available
attempt #4:fail to get shard leaders from QueryCoord: no replica available
attempt #5:fail to get shard leaders from QueryCoord: no replica available
attempt #6:fail to get shard leaders from QueryCoord: no replica available
attempt #7:fail to get shard leaders from QueryCoord: no replica available
attempt #8:context deadline exceeded
)>, <Time:{'RPC start': '2022-07-09 17:15:27.162566', 'RPC error': '2022-07-09 17:15:37.167038'}>
RPC error: [search], <BaseException: (code=1, message=fail to search on all shard leaders, err=All attempts results:
attempt #1:fail to get shard leaders from QueryCoord: no replica available
attempt #2:fail to get shard leaders from QueryCoord: no replica available
attempt #3:fail to get shard leaders from QueryCoord: no replica available
attempt #4:fail to get shard leaders from QueryCoord: no replica available
attempt #5:fail to get shard leaders from QueryCoord: no replica available
attempt #6:fail to get shard leaders from QueryCoord: no replica available
attempt #7:fail to get shard leaders from QueryCoord: no replica available
attempt #8:context deadline exceeded
)>, <Time:{'RPC start': '2022-07-09 17:15:27.159055', 'RPC error': '2022-07-09 17:15:37.167172'}>
5
```

Now our entire system is down

**Comment by yanliang567** [comment]
@SKV1983 thank you for the issue. which milvus version are you running?
Could you please refer this [script](https://github.com/milvus-io/milvus/tree/master/deployments/export-log) to export the whole Milvus logs for investigation?

/assign @SKV1983 
/unassign

**Comment by hellwhj** [comment]
我这里单机版也遇到相同报错问题
[errorlog.log.gz](https://github.com/milvus-io/milvus/files/9081262/errorlog.log.gz)

**Comment by SKV1983** [comment]
We didn't host on Kubernetes cluster, attached zip file of pulsar+etcd+minio logs.


[pulsar_minio_etcd_error_logs.zip](https://github.com/milvus-io/milvus/files/9081217/pulsar_minio_etcd_error_logs.zip)

**Comment by yanliang567** [comment]
/assign @czs007 
@czs007 could you please help to take a look?

/unassign

**Comment by czs007** [comment]
working on it .

**Comment by binbinlv** [comment]
Similar issue occurs in nightly for 2.1.0 branch:
https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-release-nightly/detail/milvus-release-nightly/154/pipeline/215

1. milvus log: [milvus-distributed-pulsar-154-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-release-nightly/154/artifact/milvus-distributed-pulsar-154-pymilvus-e2e-logs.tar.gz)
2. collection name: e2e__OuffTKtr
3. FAILED time:
[2022-07-14T23:50:41.060Z] [gw1] [ 21%] FAILED testcases/test_e2e.py::TestE2e::test_milvus_default 
4. test logs:
```
[2022-07-15T00:50:13.078Z] [2022-07-14 23:50:19 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.02853716098289656, 0.12382336215824762, 0.028840878111218222, 0.09541067850763296, 0.0752975961142934, 0.15850816577423815, 0.004079503327874338, 0.07322815509929252, 0.10480818372599095, 0.11551948515573277, 0.0018878451952638897, 0.002902037923695255, 0.05481115750146432, 0.07147565937183493,......, kwargs: {} (api_request.py:55)
[2022-07-15T00:50:13.078Z] [2022-07-14 23:50:29 - ERROR - pymilvus.decorators]: RPC error: [_execute_search_requests], <MilvusException: (code=1, message=fail to search on all shard leaders, err=All attempts results:
[2022-07-15T00:50:13.078Z] attempt #1:fail to get shard leaders from QueryCoord: no replica available
[2022-07-15T00:50:13.078Z] attempt #2:fail to get shard leaders from QueryCoord: no replica available
[2022-07-15T00:50:13.078Z] attempt #3:fail to get shard leaders from QueryCoord: no replica available
[2022-07-15T00:50:13.078Z] attempt #4:fail to get shard leaders from QueryCoord: no replica available
[2022-07-15T00:50:13.078Z] attempt #5:fail to get shard leaders from QueryCoord: no replica available
[2022-07-15T00:50:13.078Z] attempt #6:fail to get shard leaders from QueryCoord: no replica available
[2022-07-15T00:50:13.078Z] attempt #7:fail to get shard leaders from QueryCoord: no replica available
```

**Comment by SKV1983** [comment]
/assign @czs007 
@czs007, any patch ?

**Comment by zc-WUT** [comment]
> 我这里单机版也遇到相同报错问题 [errorlog.log.gz](https://github.com/milvus-io/milvus/files/9081262/errorlog.log.gz)

大佬，解决了吗

**Comment by yanliang567** [comment]
@zc-WUT which version of Milvus are you running? if 2.2.2, please file a new issue and attach all the milvus pod logs, we will follow up.

**Comment by zc-WUT** [comment]
2.1.0
&nbsp;





&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年2月1日(星期三) 下午3:07
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [milvus-io/milvus] [Bug]: fail to get shard leaders from QueryCoord: no replica available (Issue #18205)





 
@zc-WUT which version of Milvus are you running? if 2.2.2, please file a new issue and attach all the milvus pod logs, we will follow up.

**Comment by zc-WUT** [comment]
> @zc-WUT which version of Milvus are you running? if 2.2.2, please file a new issue and attach all the milvus pod logs, we will follow up.

version 2.1.4, it suddenly starts to report the same mistake as @binbinlv, no matter what collection.
